compiler/qsc_eval/src/backend.rs

```diff
@@ -419,6 +420,36 @@ impl Backend for SparseSim {
                 self.apply_noise(q);
                 Some(Ok(Value::unit()))
             }
+            "Apply" => {
+                let [matrix, qubits] = unwrap_tuple(arg);
+                let qubits = qubits.unwrap_array();
+                let qubits = qubits
```

Comment:
I know other legs of this match are also rather lengthy, but maybe there's a way to move matrix transformation into a separate function.

Reply:
Good idea, I'll see what I can refactor here!